test/component/component_test.cpp

```diff
@@ -902,3 +910,122 @@ TEST_F(ComponentTestInterface, PatchSimpleFunctionality)
 
     EXPECT_TRUE(m_callbackComplete);
 }
+
+/**
+ * @brief Test the DOWNLOAD request setting a custom user-agent.
+ */
+TEST_F(ComponentTestInterface, DownloadWithCustomUserAgent)
+{
+    const std::string userAgent {"Custom-User-Agent"};
+
+    HTTPRequest::instance().download(
+        HttpURL("http://localhost:44441/"), "./test.txt", [](auto, auto) {}, DEFAULT_HEADERS, {}, userAgent);
+
+    std::ifstream file("./test.txt");
+    std::string line;
+    std::getline(file, line);
+    EXPECT_EQ(line, "Hello World!");
+}
+
+/**
+ * @brief Test the POST request setting a custom user-agent.
+ */
+TEST_F(ComponentTestInterface, PostWithCustomUserAgent)
+{
+    const std::string userAgent {"Custom-User-Agent"};
+
+    HTTPRequest::instance().post(
+        HttpURL("http://localhost:44441/"),
+        R"({"hello":"world"})"_json,
+        [&](const std::string& /*response*/) { m_callbackComplete = true; },
+        [](auto, auto) {},
+        "",
+        DEFAULT_HEADERS,
+        {},
+        userAgent);
+
+    EXPECT_TRUE(m_callbackComplete);
+}
+
+/**
+ * @brief Test the GET request setting a custom user-agent.
+ *
+ */
+TEST_F(ComponentTestInterface, GetWithCustomUserAgent)
+{
+    const std::string userAgent {"Custom-User-Agent"};
+
+    HTTPRequest::instance().get(
+        HttpURL("http://localhost:44441/"),
+        [&](const std::string& /*response*/) { m_callbackComplete = true; },
```

Comment:
We could make use of the `/check-headers` endpoint to check if the user agent is really added to the HTTP headers:
```suggestion
        HttpURL("http://localhost:44441/check-headers"),
        [&](const std::string& response) { 
            ASSERT_EQ(nlohmann::json::parse(response).at("User-Agent"), userAgent);
            m_callbackComplete = true; 
        },
```

FYI: This is the full content of `response`, for this test:
```json
{"Accept":"application/json","Accept-Charset":"utf-8","Content-Type":"application/json","Host":"localhost:44441","REMOTE_ADDR":"::1","REMOTE_PORT":"36694","User-Agent":"Custom-User-Agent"}
```